packages/frontend/core/src/components/attachment-viewer/worker/worker.ts

```diff
@@ -0,0 +1,223 @@
+import { DebugLogger } from '@affine/debug';
+import type { Document } from '@toeverything/pdf-viewer';
+import {
+  createPDFium,
+  PageRenderingflags,
+  Runtime,
+  Viewer,
+} from '@toeverything/pdf-viewer';
+
+import type { MessageData, MessageDataType } from './types';
+import { MessageOp, RenderKind } from './types';
+
+const logger = new DebugLogger('affine:worker:pdf');
+
+let inited = false;
+let viewer: Viewer | null = null;
+let doc: Document | undefined = undefined;
+
+// Caches images with the range.
+const cached = new Map<string, ImageData>();
```

Comment:
don't forget to control size of images cached

Reply:
the front end will pass in a range, and anything outside the range will be cleared